pyproject.toml

```diff
@@ -49,3 +49,71 @@ unfixable = ["F401"]
 [tool.ruff.mccabe]
 # Unlike Flake8, default to a complexity level of 10.
 max-complexity = 10
+
+[project]
+name = "pyautogen"
+version = "0.2.2"
+description = "Enabling Next-Gen LLM Applications via Multi-Agent Conversation Framework"
+readme = "README.md"
+authors = [
+    {name = "AutoGen", email = "auto-gen@outlook.com"},
+]
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+]
+requires-python = ">=3.9, <3.13"
```

Comment:
currently, in setup.py set to:
```python
python_requires=">=3.8, <3.12"
```
To support Python 3.12, you should update github workflows first since it is not tested now.

Removing support for Python 3.8 should be discussed.

Reply:
https://github.com/microsoft/autogen/pull/1155/commits/5e9fdc3a5871ba8c5ecea23dfb444398848c33c4

Reply:
Discussion on dropping Python 3.8 support: https://github.com/microsoft/autogen/discussions/1173

Reply:
Last I check, Python 3.12 does not work. There are issues with the aiohttp library if I recall correctly. That's why it's strictly < 3.13

Reply:
updated to be <3.12